### PR TITLE
SEAB-838: noUnusedLocals set to true

### PR DESCRIPTION
**Pull request by hyunnaye**
**Description**
This PR sets `noUnusedLocals` to true.

**Review Instructions**
Unused locals should be detected.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-838

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

**Review by denis-yuen — APPROVED** [review]

File: cypress/e2e/immutableDatabaseTests/app-spec.ts [hunk 1]
```diff
@@ -31,22 +31,6 @@ describe('Logged in Dockstore Home', () => {
     cy.scrollTo('bottom');
     cy.get('[data-cy=mt-toot]').should('be.visible');
   });
-
-  function starColumn(url: string, type: string) {
```
Comment (coverbeck) [inline cypress/e2e/immutableDatabaseTests/app-spec.ts:35]: Why was this removed?

File: tsconfig.json [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// @ts-nocheck
```
Comment (coverbeck) [inline tsconfig.json:1]: What is this for? Also, TIL, there is a JSONC that supports comments in JSON. :)

**Review by coverbeck — COMMENTED** [review]
Looks good, but I had a couple of questions, left inline.

File: cypress/e2e/immutableDatabaseTests/app-spec.ts (hunk 1 above)
Comment (coverbeck) [reply cypress/e2e/immutableDatabaseTests/app-spec.ts:35]: Oh, I see, it was never invoked. Never mind.

Comment (hyunnaye) [reply cypress/e2e/immutableDatabaseTests/app-spec.ts:35]: It said it wasn't used and was catching the noUnusedLocals error

File: tsconfig.json (hunk 2 above)
Comment (hyunnaye) [reply tsconfig.json:1]: oops removed. I was testing and forgot to remove

**Review by coverbeck — APPROVED** [review]
